crates/sncast/tests/integration/data_transformer.rs

```diff
@@ -0,0 +1,1022 @@
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![Felt::from(u32::MAX).to_hex_string()];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("unsigned_fn").unwrap(),
+    )?;
+
+    let expected_output = vec![Felt::from(u32::MAX)];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_happy_case_tuple_function_cairo_expression_input() -> anyhow::Result<()> {
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![String::from("(2137_felt252, 1_u8, Enum::One)")];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("tuple_fn").unwrap(),
+    )?;
+
+    let expected_output: Vec<Felt> = vec![2137.into(), 1.into(), 0.into()];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_happy_case_tuple_function_with_nested_struct_cairo_expression_input(
+) -> anyhow::Result<()> {
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![String::from(
+        "(123, 234, Enum::Three(NestedStructWithField {a: SimpleStruct {a: 345}, b: 456 }))",
+    )];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("tuple_fn").unwrap(),
+    )?;
+
+    let expected_output: Vec<Felt> = vec![123, 234, 2, 345, 456]
+        .into_iter()
+        .map(Felt::from)
+        .collect();
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_happy_case_tuple_function_serialized_input() -> anyhow::Result<()> {
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![
+        String::from("0x859"),
+        String::from("0x1"),
+        String::from("0x0"),
+    ];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("tuple_fn").unwrap(),
+    )?;
+
+    let expected_output: Vec<Felt> = vec![2137.into(), 1.into(), 0.into()];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_happy_case_complex_function_cairo_expressions_input() -> anyhow::Result<()> {
+    let max_u256 = U256::max_value().to_string();
+
+    let simulated_cli_input = vec![
+        "array![array![0x2137, 0x420], array![0x420, 0x2137]]",
+        "8_u8",
+        "-270",
+        "\"some_string\"",
+        "(0x69, 100)",
+        "true",
+        &max_u256,
+    ]
+    .into_iter()
+    .map(String::from)
+    .collect_vec();
+
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let result = transform(
+        &simulated_cli_input,
+        contract_class,
+        &get_selector_from_name("complex_fn").unwrap(),
+    )?;
+
+    // Manually serialized in Cairo
+    let expected_output: Vec<Felt> = vec![
```

Comment:
```suggestion
    let expected_output: Vec<Felt> = [
```
in other tests can be changed too